File: ws/chiriboga/u1/ws08CDJavaClasses/ChickenFarmSimulator/src/ec/edu/espe/farmsimulator/menu/Menu.java
package ec.edu.espe.farmsimulator.menu;

import java.text.SimpleDateFormat;
import java.util.Scanner;
import static utils.FileManager.*;
import java.io.IOException;

/**
 *
 * @author Kerlly Chiriboga, ODS
 */
public class Menu {
    public static void showMenu() {
        System.out.println("Kerlly's Chicken Farm Simulator");
        try (Scanner scanner = new Scanner(System.in)) {
            SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd");
            dateFormat.setLenient(false);
            int operation;

            do {
                System.out.println("Menu:");
                System.out.println("1. Create Chicken");
                System.out.println("2. Read Chickens");
                System.out.println("3. Update Chicken");
                System.out.println("4. Delete Chicken");
                System.out.println("5. Find Chicken");
                System.out.println("6. Exit");
                System.out.println("Enter operation: ");

                operation = scanner.nextInt();
                scanner.nextLine();

                try {
                    switch (operation) {
                        case 1:
                            createChicken(scanner, dateFormat);
                            break;
                        case 2:
                            readChickens();
                            break;
                        case 3:
                            updateChicken(scanner, dateFormat);
                            break;
                        case 4:
                            deleteChicken(scanner);
                            break;
                        case 5:
                            findChicken(scanner);
                            break;
                        case 6:
                            System.out.println("Exiting...");
                            break;
                        default:
                            System.out.println("Invalid operation.");
                    }
                } catch (IOException e) {
                    System.err.println("An I/O error occurred: " + e.getMessage());
                } catch (Exception e) {
                    System.err.println("An error occurred: " + e.getMessage());
                }
            } while (operation != 6);
        }
    }
}

File: ws/chiriboga/u1/WorkshopsUnit1/ws08CDJavaClasses/ChickenFarmSimulatorPython/farm_simulator/menu/Menu.py
# File: ec/edu/espe/farmsimulator/menu/Menu.py

from farm_simulator.utils.FileManager import create_chicken, read_chickens, update_chicken, delete_chicken, find_chicken
from datetime import datetime
import sys

def show_menu():
    print("Kerlly's Chicken Farm Simulator")
    date_format = "%Y-%m-%d"
    operation = 0

    while operation != 6:
        print("Menu:")
        print("1. Create Chicken")
        print("2. Read Chickens")
        print("3. Update Chicken")
        print("4. Delete Chicken")
        print("5. Find Chicken")
        print("6. Exit")
        operation = int(input("Enter operation: "))

        try:
            if operation == 1:
                create_chicken(input, datetime.strptime)
            elif operation == 2:
                read_chickens()
            elif operation == 3:
                update_chicken(input, datetime.strptime)
            elif operation == 4:
                delete_chicken(input)
            elif operation == 5:
                find_chicken(input)
            elif operation == 6:
                print("Exiting...")
            else:
                print("Invalid operation.")
        except IOError as e:
            print(f"An I/O error occurred: {e}")
        except Exception as e:
            print(f"An error occurred: {e}")

if __name__ == "__main__":
    show_menu()
